### PR TITLE
simplify inference api

### PR DESCRIPTION
**Pull request by Superjomn**

### CODE REVIEW

File: paddle/contrib/inference/paddle_inference_api_impl.cc [hunk 1]
```diff
@@ -85,11 +84,6 @@ bool NativePaddlePredictor::Init() {
   }
   ctx_ = executor_->Prepare(*inference_program_, 0);
 
-  // Create variables
-  // TODO(panyx0718): Why need to test share_variables here?
-  if (config_.share_variables) {
-    executor_->CreateVariables(*inference_program_, scope_.get(), 0);
```
Comment (panyx0718) [inline paddle/contrib/inference/paddle_inference_api_impl.cc]: why is this deleted?

File: paddle/contrib/inference/paddle_inference_api_impl.cc [hunk 2]
```diff
@@ -124,7 +118,7 @@ bool NativePaddlePredictor::Run(const std::vector<PaddleTensor> &inputs,
                                 scope_.get(),
                                 &feed_targets,
                                 &fetch_targets,
-                                !config_.share_variables);
+                                true /*create_variable*/);
```
Comment (panyx0718) [inline paddle/contrib/inference/paddle_inference_api_impl.cc]: if share_variable is true, this should be false by default?

File: paddle/contrib/inference/paddle_inference_api_impl.cc [hunk 3]
```diff
@@ -85,11 +84,13 @@ bool NativePaddlePredictor::Init() {
...
-  // TODO(panyx0718): Why need to test share_variables here?
-  if (config_.share_variables) {
-    executor_->CreateVariables(*inference_program_, scope_.get(), 0);
-  }
+  // Create temporary variables first, so that the first batch do not need to
+  // create variables in the runtime. This is the logics of the old inference
+  // API.
+  // TODO(Superjomn) this should be modified when `Clone` is valid for
+  // multi-thread application.
+  executor_->CreateVariables(*inference_program_, scope_.get(), 0);
```
Comment (Superjomn) [inline paddle/contrib/inference/paddle_inference_api_impl.cc]: I restore the CreateVariable here. @panyx0718
